### PR TITLE
[NEXUS-8231] repository validation refinement

### PR DESCRIPTION
**Pull request by jdillon**
https://issues.sonatype.org/browse/NEXUS-8231

### CODE REVIEW

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF165-4 :white_check_mark:

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/config/ConfigurationFacetImpl.java [hunk 1]
```diff
@@ -89,29 +93,31 @@ public void validate(final Object value, final Class<?>... groups) {
...
+    ConstraintViolations.maybePropagate(violations, log);
+  }
+
+  /**
+   * Wrap attribute section value to allow configuration hierarchy to be encoded into property path.
+   */
+  private static class SectionWrapper
+  {
+    @Valid
+    private Map<String,Object> attributes = new HashMap<>(1);
```
Comment (mcculls) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/config/ConfigurationFacetImpl.java]: does the map need to be mutable? if not then `Collections.singletonMap(name, value)` would save space

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/manager/RepositoryManagerImpl.java [hunk 2]
```diff
@@ -56,10 +57,10 @@
     extends StateGuardLifecycleSupport
     implements RepositoryManager
 {
-  /**
-   * System property used to bypass installation of default repositories; useful for testing purposes.
-   */
-  private static final String SKIP_DEFAULT_REPOSITORIES = "skipDefaultRepositories";
+  // FIXME: Use prefixed property
+
+  private static final boolean skipDefaultRepositories =
```
Comment (mcculls) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/manager/RepositoryManagerImpl.java]: upper-case for static constant?

**Comment by mcculls** [comment]
+1

File: components/nexus-common/src/test/java/org/sonatype/nexus/common/validation/BeanValidationTrial.groovy [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+import javax.validation.Valid
+import javax.validation.Validation
+import javax.validation.ValidatorFactory
+import javax.validation.constraints.NotNull
+
+/**
+ * Bean validation trials.
+ */
+class BeanValidationTrial
```
Comment (kellyrob99) [inline components/nexus-common/src/test/java/org/sonatype/nexus/common/validation/BeanValidationTrial.groovy]: what's the purpose of the 'trial's in this module? They don't run as part of the build, and seem to favor println over assertions

**Comment by kellyrob99** [comment]
+1

**Comment by mrprescott** [comment]
+1

File: components/nexus-common/src/test/java/org/sonatype/nexus/common/validation/BeanValidationTrial.groovy (hunk 3 above)
Comment (jdillon) [inline components/nexus-common/src/test/java/org/sonatype/nexus/common/validation/BeanValidationTrial.groovy]: trials are just playgrounds to test out how code behaves using junit to drive the execution.

**Comment by jdillon** [comment]
manually merged

**Pull request closed without merging**